Automattic/jetpack
# Project build/test scripts shouldn't install

#### Changes proposed in this Pull Request:

A `composer install` seems to take around 3 seconds, and a `pnpm install`
around 4. When we have 66 projects to build, that can add up.
    
We already have `jetpack build` doing the installs, so let's remove all
the install calls from all the "build-production" and
"build-development" scripts. That also means we need to remove any
`rm -rf vendor/` and the like from those scripts. It also means we have
to add logic to `jetpack build` to pass
`-o --no-dev --classmap-authoritative --prefer-dist` when doing
`composer install` for plugins.

Then let's do the same for test scripts, so there's less chance people
will get confused by different requirements for the different kinds of
scripts.

While we're at it, let's also remove the need for boilerplate
`Composer\Config::disableProcessTimeout` too by passing `--timeout=0`
when the Jetpack CLI calls composer for build and tests.
    
And let's clean up all the deprecated Jetpack build stuff out of the
monorepo root package.json, mainly because I don't want to bother to
figure out which pieces I'm breaking by doing the above.

#### Jetpack product discussion


p9dueE-4dJ-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* The build artifact shouldn't reflect any actual changes to the mirrors. Check that.
* Check that `jetpack build` and `jetpack test` work as expected locally.
